### DISCUSSION TITLE
HTML5, Semantica e SEO

### DISCUSSION THREAD

**Discussion by thismarcoantonio** [frontendbr/forum#1115] [Perguntas e Respostas]
Quando falamos em `section`, `article` e outras tags do HTML5, sempre é recomendável começar fazendo um nó com o `h1`.

Ex: 
```
<section>
  <h1>Hello World</h1>
</section>
```

Porém como o H1 significa um heading, qual o impacto para textos extensos no mesmo? Tenho um componente cujo possui um texto inicial de 20 palavras, seria incorreto semanticamente falando usar este mesmo em um h1? Se sim, seria melhor eu abandonar a semantica neste componente e partir para uma `div` com `span`?

**Comment by tarciozemel** [comment]
Semântica: até onde eu sei, não há limitações técnicas quanto ao número de caracteres. Se semanticamente ele realmente for um título, tudo bem.

SEO: salvo engano, o pessoal recomenda um limite de até 60 caracteres.

**Comment by thismarcoantonio** [comment]
@tarciozemel neste caso, possuo 117 caracteres. Alguma sugestão de como estruturar isso?

Até o momento substitui por uma div e ignorei toda semantica neste componente.

**Comment by ninetails** [comment]
Pessoalmente falando: se é texto extenso e não tem título, é um parágrafo, não? Deveria ser um conteúdo referenciado por alguma estrutura pai com um título específico.

Quando escrevo html e penso na semântica, sempre penso que o conteúdo deve fazer sentido, algo como escrever um livro. E os títulos são necessários paralelamente como se fossem sumários.

**Comment by thismarcoantonio** [comment]
@ninetails penso da mesma forma, porém não posso fazer nada, apenas executando layout :/
De qualquer forma valeu pela dica, esse componente é pra ser um "sub hero", porém n tem titulo, apenas dois grandes paragrafos. Apenas ignorei SEO e semantica.

**Comment by ninetails** [comment]
@thismarcoantonio bom, diria que não faz sentido ele ser um section dado que o hero vai ter o título e este seria um parágrafo dentro do hero. Não seria ignorar semântica, e sim ele sempre depender de um "hero pai".

Porém, se ele realmente precisar ser um section/componente, sem um pai realmente definido, diria pra botar um título com `.sr-only` então.
